lib/Interpreter/CppInterOp.cpp

```diff
@@ -718,31 +719,48 @@ namespace Cpp {
     return ComputeBaseOffset(getSema().getASTContext(), DCXXRD, Paths.front());
   }
 
-  // FIXME: We should make the std::vector<TCppFunction_t> an out parameter to
-  // avoid copies.
-  std::vector<TCppFunction_t> GetClassMethods(TCppScope_t klass)
-  {
+  void GetClassMethods(TCppScope_t klass,
+                       std::vector<TCppFunction_t>& methods) {
+    if (klass) {
+      auto* D = (clang::Decl*)klass;
 
-    if (!klass)
-      return {};
+      if (auto* TD = dyn_cast<TypedefNameDecl>(D))
+        D = GetScopeFromType(TD->getUnderlyingType());
+
+      if (D && isa<CXXRecordDecl>(D)) {
+        auto* CXXRD = dyn_cast<CXXRecordDecl>(D);
+        getSema().ForceDeclarationOfImplicitMembers(CXXRD);
+        for (Decl* DI : CXXRD->decls()) {
+          if (auto* MD = dyn_cast<CXXMethodDecl>(DI))
+            methods.push_back(MD);
+          else if (auto* USD = dyn_cast<UsingShadowDecl>(DI))
+            if (auto* MD = dyn_cast<CXXMethodDecl>(USD->getTargetDecl()))
+              methods.push_back(MD);
+        }
+      }
+    }
+  }
 
-    auto *D = (clang::Decl *) klass;
+  void GetFunctionTemplatedDecls(TCppScope_t klass,
+                                 std::vector<TCppFunction_t>& methods) {
+    if (klass) {
```

Comment:
We need to invert the flow to reduce the indentation and improve clarity. We need:
```cpp
if (!klass)
    return;
```

Reply:
done